utility/database/check_can_associate_unique_ids.go

```diff
@@ -6,55 +6,61 @@ import (
 	"github.com/PretendoNetwork/nex-go/v2"
 	"github.com/PretendoNetwork/nex-go/v2/types"
 	common_globals "github.com/PretendoNetwork/nex-protocols-common-go/v2/globals"
+	utility_types "github.com/PretendoNetwork/nex-protocols-go/v2/utility/types"
 )
 
-// This handles all relevant validity checking (including the existence of an id, if the password matches, etc)
-func CheckCanAssociateUniqueIDs(manager *common_globals.UtilityManager, userPid types.PID, uniqueIds, passwords types.List[types.UInt64]) *nex.Error {
-	var uniqueId, dbPassword types.UInt64
-	var associatedPid types.PID
+// CheckCanAssociateUniqueIDs handles all relevant validity checking (including the existence of a unique ID, if the password matches, etc)
+func CheckCanAssociateUniqueIDs(manager *common_globals.UtilityManager, userPID types.PID, uniqueIDInfos types.List[utility_types.UniqueIDInfo]) *nex.Error {
+	var uniqueID, dbPassword types.UInt64
+	var associatedPID types.PID
+
+	uniqueIDs := make([]types.UInt64, 0)
```

Comment:
You want to use our `types.List` type, otherwise the database query won't work out of the box

```suggestion
	uniqueIDs := make(types.List[types.UInt64], 0)
```